diff --git a/src/backends/interfaces.cpp b/src/backends/interfaces.cpp
index 36bf824..72cf615 100644
--- a/src/backends/interfaces.cpp
+++ b/src/backends/interfaces.cpp
@@ -22,6 +22,8 @@
 #include <format>
 #include <algorithm>
 
+#include "../config.h"
+
 namespace ASGenerator
 {
 
@@ -104,7 +106,8 @@ std::string Package::toString() const
 
 std::string PackageIndex::dataPrefix() const
 {
-    return "/usr";
+    const auto &conf = Config::get();
+    return conf.dataPrefix.empty() ? "/usr" : conf.dataPrefix;
 }
 
 } // namespace ASGenerator
diff --git a/src/config.cpp b/src/config.cpp
index 1ee92d8..da0a3c1 100644
--- a/src/config.cpp
+++ b/src/config.cpp
@@ -252,6 +252,11 @@ void Config::loadFromFile(
     if (caInfoNode)
         caInfo = Yaml::nodeStrValue(caInfoNode);
 
+    // optional override for the installation prefix of packages in this archive
+    auto dataPrefixNode = Yaml::nodeByKey(root, "DataPrefix");
+    if (dataPrefixNode)
+        dataPrefix = Yaml::nodeStrValue(dataPrefixNode);
+
     // allow specifying the AppStream format version we build data for.
     auto formatVersionNode = Yaml::nodeByKey(root, "FormatVersion");
     if (formatVersionNode) {
diff --git a/src/config.h b/src/config.h
index ec5d5fe..d20dd26 100644
--- a/src/config.h
+++ b/src/config.h
@@ -124,6 +124,7 @@ public:
 
     std::string backendName;
     Backend backend;
+    std::string dataPrefix;
     std::vector<Suite> suites;
     std::vector<std::string> oldsuites;
     DataType metadataType;
